core/commands/IssueAddCommand.php

```diff
@@ -135,16 +135,27 @@ function validate() {
 
 		$t_project_id = mci_get_project_id( $t_issue['project'] );
 
-		if( $t_project_id == ALL_PROJECTS ) {
+		if( $t_project_id == ALL_PROJECTS &&
+			!isset( $t_issue['project']['id'] ) && !isset( $t_issue['project']['name'] ) ) {
 			throw new ClientException(
 				'Project not specified',
 				ERROR_EMPTY_FIELD,
 				array( 'project' ) );
 		}
 
 		if( !project_exists( $t_project_id ) ) {
+			if( $t_project_id == ALL_PROJECTS ) {
+				if( isset( $t_issue['project']['id'] ) && !is_blank( isset( $t_issue['project']['id'] ) ) ) {
```

Comment:
I don't understand what you want with`!is_blank( isset ...`
Test if a bolean is not blank?